src/sqlfluff/dialects/dialect_greenplum.py

```diff
@@ -167,13 +248,77 @@ class CreateTableStatementSegment(postgres.CreateTableStatementSegment):
             ),
             Sequence("TABLESPACE", Ref("TablespaceReferenceSegment")),
+            Ref("DistributedBySegment")
+        ),
+    )
+
+
+class CreateTableAsStatementSegment(postgres.CreateTableAsStatementSegment):
+    """A `CREATE TABLE AS` statement.
+
+    As specified in
+    https://docs.vmware.com/en/VMware-Tanzu-Greenplum/6/greenplum-database/GUID-ref_guide-sql_commands-CREATE_TABLE_AS.html
+    This is overriden from Postgres to add the `DISTRIBUTED` clause.
+    """
+
+    match_grammar = Sequence(
+        "CREATE",
+        OneOf(
             Sequence(
-                "DISTRIBUTED",
-                OneOf(
-                    "RANDOMLY",
-                    "REPLICATED",
-                    Sequence("BY", Bracketed(Ref("ColumnReferenceSegment"))),
+                OneOf("GLOBAL", "LOCAL", optional=True),
+                Ref("TemporaryGrammar"),
+            ),
+            "UNLOGGED",
+            optional=True,
+        ),
+        "TABLE",
+        Ref("IfNotExistsGrammar", optional=True),
+        Ref("TableReferenceSegment"),
+        AnyNumberOf(
+            Sequence(
+                Bracketed(
+                    Delimited(Ref("ColumnReferenceSegment")),
+                ),
+                optional=True,
+            ),
+            Sequence("USING", Ref("ParameterNameSegment"), optional=True),
+            OneOf(
+                Sequence(
+                    "WITH",
+                    Bracketed(
+                        Delimited(
+                            Sequence(
+                                Ref("ParameterNameSegment"),
+                                Sequence(
+                                    Ref("EqualsSegment"),
+                                    OneOf(
+                                        Ref("LiteralGrammar"),
+                                        Ref("NakedIdentifierSegment"),
+                                    ),
+                                    optional=True,
+                                ),
+                            )
+                        )
+                    ),
                 ),
+                Sequence("WITHOUT", "OIDS"),
+                optional=True,
             ),
+            Sequence(
+                "ON",
+                "COMMIT",
+                OneOf(Sequence("PRESERVE", "ROWS"), Sequence("DELETE", "ROWS"), "DROP"),
+                optional=True,
+            ),
+            Sequence("TABLESPACE", Ref("TablespaceReferenceSegment"), optional=True),
+        ),
+        "AS",
+        OneOf(
+            OptionallyBracketed(Ref("SelectableGrammar")),
+            OptionallyBracketed(Sequence("TABLE", Ref("TableReferenceSegment"))),
+            Ref("ValuesClauseSegment"),
+            OptionallyBracketed(Sequence("EXECUTE", Ref("FunctionSegment"))),
         ),
+        Ref("WithDataClauseSegment", optional=True),
+        Ref("DistributedBySegment"),
```

Comment:
this is probably your issue - this wants to be optional

Reply:
I sorted this.